homeassistant/components/modbus/__init__.py

```diff
@@ -189,6 +191,15 @@
                 CONF_SWAP_WORD_BYTE,
             ]
         ),
+        vol.Optional(CONF_REGISTER_SIZE_BYTES, default=RegisterBytes.NOT_SET): vol.In(
+            [
+                RegisterBytes.NOT_SET,
```

Comment:
You would reduce the code a lot, but making it a positive_int, and then have a validator control the values.

With the enum in constants.py, you (as maintainer) need to check if the allowed input is a string (which I thought) or an integer.

Reply:
Ok though you had asked for it to be an enum in an earlier comment

Reply:
That was if you wanted to use strings not numbers.

The enum might be acceptable, but together with the option NOT_SET, means you have a lot of not needed "if" in the code.